### PR TITLE
Add constructor functions for Handler and Transport

### PR DESCRIPTION
**Pull request by ramonza**
These set the default propagation format to TraceContext.

### CODE REVIEW

File: plugin/ochttp/client.go [hunk 1]
```diff
@@ -69,7 +77,7 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	// TODO: remove excessive nesting of http.RoundTrippers here.
 	format := t.Propagation
 	if format == nil {
-		format = defaultFormat
+		format = compatibilityFormat
```
Comment (rakyll) [inline plugin/ochttp/client.go]: Why this change?

File: plugin/ochttp/client.go [hunk 2]
```diff
@@ -25,9 +25,7 @@ import (
 // Transport is an http.RoundTripper that instruments all outgoing requests with
 // OpenCensus stats and tracing.
 //
-// The zero value is intended to be a useful default, but for
-// now it's recommended that you explicitly set Propagation, since the default
-// for this may change.
+// Use NewTransport to create a Transport with the default configuration.
```
Comment (rakyll) [inline plugin/ochttp/client.go]: I think this is a confusing choice. Zero-values are expected to support the default configuration.

We should deprecate the use of Transport zero value if this is the way to go.

File: plugin/ochttp/server.go [hunk 3]
```diff
@@ -68,6 +72,17 @@ type Handler struct {
...
 }
 
+// NewHandler wraps the given handler to add stats and tracing instrumentation.
+// By default, W3C TraceContext propagation format is used and the returned
+// handler is assumed to be a public endpoint.
+func NewHandler(next http.Handler) *Handler {
+	return &Handler{
+		Handler:          next,
+		Propagation:      defaultFormat,
+		IsPublicEndpoint: true,
```
Comment (rakyll) [inline plugin/ochttp/server.go]: It is less likely that a server is a a public endpoint.

File: plugin/ochttp/client.go (hunk 1 above)
Comment (ramonza) [reply plugin/ochttp/client.go]: I just renamed the variable `defaultFormat` to `compatibilityFormat` to make it clear that we intend the new default to be TraceContext.

File: plugin/ochttp/client.go (hunk 2 above)
Comment (ramonza) [reply plugin/ochttp/client.go]: I would like to deprecate use of the zero value. Just not exactly sure how to go about this, except for what I have done here which is to document that NewTransport should be used.

**Comment by ramonza** [comment]
Waiting for: https://github.com/census-instrumentation/opencensus-specs/pull/185

File: plugin/ochttp/server.go (hunk 3 above)
Comment (ramonza) [reply plugin/ochttp/server.go]: I would guess that it's really a lot *more* likely to be public for most users, especially the smaller users that are not running complex microservice architectures. It is also a safer default.

**Comment by ramonza** [comment]
Closing because I believe we decided to make a backwards incompatible change to TraceContext instead.
